packages/prisma-client/prisma/migrations/20220909124542_move_to_builds/migration.ts

```diff
@@ -0,0 +1,44 @@
+import { PrismaClient, Prisma } from "./client";
+import { v4 as uuidv4 } from "uuid";
+
+export default () => {
+  const client = new PrismaClient();
+  return client.$transaction(async (prisma) => {
+    const trees = await prisma.tree.findMany();
+    const projects = await prisma.project.findMany();
+
+    const builds: Prisma.BuildCreateManyInput[] = [];
+
+    for (const tree of trees) {
+      const project = projects.find(
+        (project) =>
+          project.devTreeId === tree.id ||
+          project.prodTreeId === tree.id ||
+          JSON.parse(project.prodTreeIdHistory).includes(tree.id)
```

Comment:
I would also run it over zod schema to be sure

Reply:
I mean in all cases where we JSON.parse or stringify

Reply:
This is a bit tricky. All code that is used in a migration has to be immutable to changes that we make to the rest of the project. We can't just import something. We would have to copy everything we want to use into the migration folder.

Reply:
oh that's an interesting point, definitely need to document that! maybe even add to each migration file as a comment and in a template, because this is SUPER non-obvious

Reply:
I guess we could copy the schema we need to the local modules, just like the client

Reply:
> maybe even add to each migration file as a comment and in a template

Good idea! Will add to the template

> I guess we could copy the schema we need to the local modules, just like the client

Something to keep in mind, but for now will just copy Zod definitions to the migration file manually

Reply:
that's what I meant, zod schema definitions, copied, could be separate files next to the migration